tests/unit_tests/libs_tests/ECS_tests/Systems_tests/Death_System_tests.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+** EPITECH PROJECT, 2022
+** R-Type
+** File description:
+** Death_System_tests
+*/
+
+
+#include <criterion/criterion.h>
+#include "World/World.hpp"
+#include "Entity/Entity.hpp"
+#include "GameSystems/DeathSystem.hpp"
+#include "GameEntityManipulation/CreateEntitiesFunctions/CreateProjectile.hpp"
+#include "GameComponents/DeathComponent.hpp"
+#include "GameComponents/ProjectileComponent.hpp"
+
+using namespace ecs;
+
+Test(Death_system, kill_entity)
+{
+    World world(1);
+
+    createNewProjectile(world, 10, 10, 1, 1, 10);
+
+    std::vector<std::shared_ptr<ecs::Entity>> joined = world.joinEntities<Projectile>();
+
+    world.getEntity(1).addComponent<Death>();
+
+    world.addSystem<DeathSystem>();
+
+    world.runSystems();
+
+    std::vector<std::shared_ptr<ecs::Entity>> death_joined = world.joinEntities<Projectile>();
+
+    cr_assert_eq(1, joined.size());
+    cr_assert_eq(0, death_joined.size());
+}
+
+Test(Death_system, kill_entities)
+{
+    World world(1);
+
+    createNewProjectile(world, 10, 10, 1, 1, 10);
+    createNewProjectile(world, 10, 10, 1, 1, 10);
+    createNewProjectile(world, 10, 10, 1, 1, 10);
+
+    std::vector<std::shared_ptr<ecs::Entity>> joined = world.joinEntities<Projectile>();
+
+    world.getEntity(2).addComponent<Death>();
+
+    world.addSystem<DeathSystem>();
+
+    world.runSystems();
+
+    std::vector<std::shared_ptr<ecs::Entity>> death_joined = world.joinEntities<Projectile>();
+
+    cr_assert_eq(3, joined.size());
+    cr_assert_eq(2, death_joined.size());
+}
```

Comment:
Did you test when no Death component is inside the entities ? And when no entities is in the world ?

Reply:
![Screenshot from 2022-10-13 02-11-56](https://user-images.githubusercontent.com/72011124/195470367-2fe6e32d-c806-46a7-a460-64cf9bc21c49.png)